youtube_dl/extractor/rockstargames.py

```diff
@@ -0,0 +1,55 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    qualities,
+    parse_iso8601
+)
+
+
+class RockstarGamesIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?rockstargames\.com/videos/video/(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'https://www.rockstargames.com/videos/video/11544/',
+        'md5': '03b5caa6e357a4bd50e3143fc03e5733',
+        'info_dict': {
+            'id': '11544',
+            'ext': 'mp4',
+            'title': 'Further Adventures in Finance and Felony Trailer',
+            'thumbnail': 're:^https?://.*\.jpg$',
+            'description': """Build an empire as a CEO. Traffic illicit cargo and contraband. Disrupt rival supply chains. Acquire extravagant new vehicles. Open prestigious new offices. Hire an executive assistant. And run the town, one hostile takeover at a time.""",
+            'upload_date': '20160602',
+            'timestamp': 1464876000
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        json_url = 'https://www.rockstargames.com/videoplayer/videos/get-video.json?id=%s&locale=en_us' % (video_id)
+        json_data = self._download_json(json_url, video_id)['video']
+
+        formats = []
+        get_quality = qualities(['360p', '720p', '1080p'])
+
+        for video in json_data['files_processed']['video/mp4']:
+            if not video.get('src'):
+                continue
+            video_format = {
+                'url': self._proto_relative_url(video['src']),
+                'quality': get_quality(video.get('resolution')),
```

Comment:
And done )

On Mon, Jun 13, 2016 at 7:32 PM, Sergey M. <notifications@github.com> wrote:

> In youtube_dl/extractor/rockstargames.py
> <https://github.com/rg3/youtube-dl/pull/9737#discussion_r66833229>:
>
> > +    }
> > +
> > +    def _real_extract(self, url):
> > +        video_id = self._match_id(url)
> > +        json_url = 'https://www.rockstargames.com/videoplayer/videos/get-video.json?id=%s&locale=en_us' % (video_id)
> > +        json_data = self._download_json(json_url, video_id)['video']
> > +
> > +        formats = []
> > +        get_quality = qualities(['360p', '720p', '1080p'])
> > +
> > +        for video in json_data['files_processed']['video/mp4']:
> > +            if not video.get('src'):
> > +                continue
> > +            video_format = {
> > +                'url': self._proto_relative_url(video['src']),
> > +                'quality': get_quality(video.get('resolution')),
>
> Instead of using quality parse height from resolution and put as height
> field.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/rg3/youtube-dl/pull/9737/files/8476911650737551f2e4c331aca1f8b65b507e61#r66833229>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC-3KHEO4HWzsMWUyuRMg3wvO8EMeuVUks5qLZQjgaJpZM4Ix4UC>
> .
>